internal/vm/setup.go

```diff
@@ -62,40 +62,40 @@ _G["process_auto_fields"] = function(plugin)
   return json.encode(p)
 end
 
-_G["validate_plugin_schema"] = function(plugin_schema_string)
+local function load_plugin_schema(plugin_schema_string, validate_plugins)
   local plugin_schema = loadstring(plugin_schema_string)()
   if plugin_schema == nil then
     return nil, "invalid plugin schema: cannot be empty"
   end
-
   local ok, err_t = MetaSchema.MetaSubSchema:validate(plugin_schema)
   if not ok then
     return nil, tostring(errors:schema_violation(err_t))
   end
-
   local plugin_name = plugin_schema.name
-  ok, err = Entity.new_subschema(validate_plugins, plugin_name, plugin_schema)
+  local ok, err = Entity.new_subschema(validate_plugins, plugin_name, plugin_schema)
   if not ok then
     return nil, "error loading schema for plugin: " .. err
   end
-  validate_plugins.subschemas[plugin_name] = nil
+  return plugin_schema, nil
+end
 
+_G["validate_plugin_schema"] = function(plugin_schema_string)
+  local validate_plugins = assert(Entity.new(plugins_definition))
+  local plugin_schema, err = load_plugin_schema(plugin_schema_string, validate_plugins)
+  if err ~= nil then
+    return nil, err
+  end
+  local plugin_name = plugin_schema.name
+  validate_plugins.subschemas[plugin_name] = nil
```

Comment:
@mikefero , since I changed the validate plugins to a local variable within a function, so I am not sure if this action is required for setting the subschemas to nil. Thought I'd check